aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -1880,6 +1880,27 @@ Tensor masked_scatter(const Tensor & self, const Tensor & mask, const Tensor & s
   return _self->clone(at::MemoryFormat::Contiguous).masked_scatter_(*_mask, source);
 }
 
+Tensor masked_scatter_backward_symint(
```

Comment:
cc @zou3519 @albanD @ezyang 

Previously masked_select_backward was CompositeImplicit, which had the disadvantage of materializing a dynamic shape tensor even though the shapes on the overall op are static. 

What do you think about moving the backward to an op?

Reply:
That's ok to do but you will need to add double backward formula manually and make sure you don't regress anything.

Reply:
That's ok to do but you will need to add double backward formula manually and make sure you don't regress anything.

Reply:
That's ok to do but you will need to add double backward formula manually and make sure you don't regress anything.

Reply:
Is there a preference for aten changes vs inductor only changes ?

Reply:
I can't really say as I'm not sure what the inductor change would entail.
But making the backward an op with it's own derivative is always faster. So no concern on my end to go with that. It is a bit of work to properly come up with these derivatives and implement them properly.

Reply:
I bet the double backwards formula is ultimately preferable